activerecord/lib/active_record/type/decimal.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module Type
+    class Decimal < ActiveModel::Type::Decimal # :nodoc:
+      private
+
+        def cast_value(value)
+          raise ActiveModel::RangeError, "cannot be infinite" if value.to_d.infinite?
```

Comment:
Can we move the check to the `ensure_in_range`? I don't think we should try to convert the value to decimal twice.

BTW would not `ensure_in_range` already capture this since Infinity is not covered by any range?

Reply:
`ensure_in_range` is defined on the integer type, so it's not called here.

I do think we should implement an equivalent method for the decimal type though.

Reply:
I thought of this, but I wasn't sure how to get the max and min numbers we want to support. 1 +- Infinity is still Infinity.